src/raftstore/store/peer.rs

```diff
@@ -661,21 +635,20 @@ impl Peer {
                     // the first empty entry on its term. After that the lease expiring time
                     // should be updated to
                     //   send_to_quorum_ts + max_lease
-                    // as the comments in `next_lease_expired_time` function explain.
+                    // as the comments in `Lease` explain.
                     // It is recommended to update the lease expiring time right after
                     // this peer becomes leader because it's more convenient to do it here and
                     // it has no impact on the correctness.
-                    let next_expired_time = self.next_lease_expired_time(monotonic_raw_now());
-                    self.leader_lease_expired_time = Some(Either::Left(next_expired_time));
+                    self.renew_leader_lease(monotonic_raw_now());
```

Comment:
It should be updated even when the lease is under suspected stage.

Reply:
It will update, as long as `monotonic_raw_now() + max_lease` is later than the recorded lease expired time.